### PR TITLE
[trie] test leaf node

### PR DESCRIPTION
**Pull request by huof6829**
# Description

Fixes https://github.com/iotexproject/iotex-core/issues/3202

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [] My code follows the style guidelines of this project
- [] I have performed a self-review of my code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [] New and existing unit tests pass locally with my changes
- [] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by huangzhiran — APPROVED** [review]

File: db/trie/mptrie/leafnode_test.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+func TestLeafBase(t *testing.T) {
+	var (
+		require = require.New(t)
+		cli     = &merklePatriciaTrie{
+			hashFunc: DefaultHashFunc,
+			kvStore:  trie.NewMemKVStore(),
+		}
+		key1       = keyType{1, 2, 3, 4, 5}
+		expectKeys = []keyType{
```
Comment (Liuhaai) [inline db/trie/mptrie/leafnode_test.go:128]: key could be created from string

File: db/trie/mptrie/leafnode_test.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+	lnode, err := newLeafNode(cli, keyType("iotex"), []byte("coin"))
+	require.NoError(err)
+
+	for i, item := range items {
+		node, err := lnode.Upsert(cli, item.k, offset, item.v)
+		require.NoError(err)
+
+		switch i {
+		case 0:
```
Comment (Liuhaai) [inline db/trie/mptrie/leafnode_test.go:93]: The usage `case 0, 1, 2, ...` is strange in for loop

File: db/trie/mptrie/leafnode_test.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	nodepb, ok := proto.(*triepb.NodePb)
+	require.True(ok)
+	leaf, ok := nodepb.Node.(*triepb.NodePb_Leaf)
+	require.True(ok)
+	lnode1 := newLeafNodeFromProtoPb(leaf.Leaf, nil)
+	require.Equal(lnode.key, lnode1.key)
+	require.Equal(lnode.value, lnode1.value)
+}
+
+func TestLeafNodeDelete(t *testing.T) {
```
Comment (Liuhaai) [inline db/trie/mptrie/leafnode_test.go:35]: test cases in `TestLeafNodeUpsert` could be used in `TestLeafNodeDelete`

**Review by Liuhaai — APPROVED** [review]

File: db/trie/mptrie/leafnode_test.go (hunk 3 above)
Comment (huof6829) [reply db/trie/mptrie/leafnode_test.go]: Fixed

File: db/trie/mptrie/leafnode_test.go (hunk 2 above)
Comment (huof6829) [reply db/trie/mptrie/leafnode_test.go]: fixed

File: db/trie/mptrie/leafnode_test.go (hunk 1 above)
Comment (huof6829) [reply db/trie/mptrie/leafnode_test.go]: ok

**Pull request merged**